include/systems/implicit_system.h

```diff
@@ -352,6 +357,11 @@ class ImplicitSystem : public ExplicitSystem
   bool zero_out_matrix_and_rhs;
 
 
+  /**
+   * Initializes the matrices associated with this system.
+   */
+  virtual void init_matrices ();
```

Comment:
Maybe I didn't fully understand the new design that was discussed. Did `init_matrices()` need to become public?